docker-compose-kupo.yml

```diff
@@ -0,0 +1,63 @@
+version: '3.8'
+services:
+  server:
+    image: ghcr.io/geniusyield/dex-contracts-api:latest
+    container_name: server
+    restart: always
+    volumes: ["node-ipc:/ipc"]
+    ports: ["8082:8082"]
+    depends_on: { node: { condition: service_healthy } }
+    environment:
+      MAESTRO_API_KEY: ${MAESTRO_API_KEY}
+      SERVER_API_KEY: ${SERVER_API_KEY}
+      SEED_PHRASE: ${SEED_PHRASE}
+      SERVER_CONFIG: |
+        coreProvider:
+          socketPath: /ipc/node.socket
+          kupoUrl: "kupo:1442"
+        networkId: mainnet
+        logging: [{type: {tag: stderr}, severity: "Debug", verbosity: "V2"}]
+        maestroToken: <<MAESTRO_API_KEY>>
+        serverApiKey: <<SERVER_API_KEY>>
+        wallet:
+          tag: mnemonicWallet
+          contents:
+            mnemonic: <<SEED_PHRASE>>
+        port: 8082
+  kupo:
+    image: cardanosolutions/kupo
+    container_name: kupo
+    restart: always
+    volumes: ["node-ipc:/ipc", "./cardano-configurations/network:/network"]
+    ports: ['1442:1442']
+    depends_on: { node: { condition: service_healthy } }
+    command: [
+     --node-socket, /ipc/node.socket,
+     --since, origin,
+     --match, "*",
+     --host, "0.0.0.0",
+     --node-config, "/network/${NETWORK-mainnet}/cardano-node/config.json",
+     --in-memory,
```

Comment:
Not sure if `in-memory` is appropriate. Also, is there a flexibility for user to specify their own kupo configuration?

Reply:
This is just an example. Anybody can clone the repository and change this.

It is intended to showcase how you can configure the system so it is using Kupo.

In the readme we can add some notes about this, in case you have some suggestions.

Reply:
The given configuration is building the index of entire UTxO set in memory, this may cause issue for many hence the concern. `--workdir` is what I would in general prefer but we can go with what is presented in this PR for now.
<img width="1119" alt="image" src="https://github.com/geniusyield/dex-contracts-api/assets/29835247/7e85abcb-194d-4eca-911a-c87a38b7c0df">

Reply:
We can add a note about the recommendation to consider using workdir in production.

Reply:
I have changed it to --workdir in 9b2883038e50b4ed2cc2fb31b083f84a48fbeeb6